web/pages/[username]/[contractSlug].tsx

```diff
@@ -227,7 +227,7 @@ function ContractLeaderboard(props: { contract: Contract; bets: Bet[] }) {
         setUsers(sortedUsers)
       })
     }
-  }, [])
+  }, [top5Ids, userProfits])
```

Comment:
As Austin is cautioning, this addition to the useEffect dependencies is not good.

top5Ids and userProfits are new objects on each render, so it will trigger a the effect every time, which will call our listUsers api, which will set state, which will trigger another render...

I think this is the only deps that's a problem in this change, but I would be very careful in the future.

Reply:
Wow, I guess I was confused, I assumed the dependencies were doing some kind of deep equality check instead of reference equality...I will fix this up to not be wrong. This is my first project using React hooks so I have some things to learn :/

Reply:
Thanks for the catch, James!

@mqp can we add a comment here that surpresses the ESLint warning and tells other code viewers to be very careful while editing this line (maybe referencing this comment thread?) Thanks!